core/tx_pool.go

```diff
@@ -220,11 +223,12 @@ type TxPool struct {
 	wg sync.WaitGroup // for shutdown sync
 
 	homestead bool
+	EnodeId   string
 }
 
 // NewTxPool creates a new transaction pool to gather, sort and filter inbound
 // transactions from the network.
-func NewTxPool(config TxPoolConfig, chainconfig *params.ChainConfig, chain blockChain) *TxPool {
+func NewTxPool(config TxPoolConfig, chainconfig *params.ChainConfig, chain blockChain, id string) *TxPool {
```

Comment:
Is there a better way to provide the id without changing the method signature. That would bring worry-free when merging from upstream.

Suggestion: how about move it to `TxPoolConfig`?

Reply:
Based on the usage of `EnodeId` inside `TxPool`, only  `api_backend.go:func (b *EthAPIBackend) SendTx(..)` uses it to perform validation. 

IMO, `enodeId` should be referenced rather than constructed (https://github.com/jpmorganchase/quorum/blob/e9a05f63e7aee9cb82b7f34042b9fd171bd05c92/eth/backend.go#L200) and passed around 

`p2p.Server` is the right struct to reference the `enodeId` via its `LocalNode().ID()`. However, there is no reference to `p2p.Server` from `EthAPIBackend`. I did a quick check upstream and found that 
https://github.com/ethereum/go-ethereum/blob/v1.9.0/eth/backend.go#L66-L72

Adding `p2p.Server` to `Ethereum` which then can be referenced from `EthAPIBackend` to fulfill the validation needs